envoy/router/router.h

```diff
@@ -724,6 +724,38 @@ using TlsContextMatchCriteriaConstPtr = std::unique_ptr<const TlsContextMatchCri
  */
 class HttpRouteTypedMetadataFactory : public Envoy::Config::TypedMetadataFactory {};
 
+/**
+ * Base class for all early data option extensions.
+ */
+class EarlyDataOption {
+public:
+  virtual ~EarlyDataOption() = default;
+
+  /**
+   * @return bool whether the given request should be allowed to be sent over early data.
+   */
+  virtual bool allowsEarlyDataForRequest(Http::RequestHeaderMap& request_headers) const PURE;
```

Comment:
Should this be a const ref, or do we expect implementations to mutate |request_headers|?